eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,573 @@
+    if dummification:
+        tmp = pd.get_dummies(join_grid[join_grid[target].isin(identified_values)][[target]], prefix="", prefix_sep="")
+        indicators = tmp.groupby(tmp.index).max()
+    else:
+        tmp = join_grid[target]
+        indicators = tmp.groupby(tmp.index).max()
+    return indicators, join_grid
+
+
+@beartype
+def _check_and_prepare_param(
+    geodata: gpd.GeoDataFrame,
+    column: str,
+    subset_target_attribute_values: list,
+    target_name: Optional[str] = "Added",
+) -> Tuple[bool, str, list]:
+    """Intermediate utility.
+
+    Prepare the parameters and check their consistency.
+
+    Args:
+        geodata: Geodata object corresponding to the vector file
+        column: Name of the column of interest. If no attribute is
+            specified, a synthetic one is produced to represent the
+            absence/presence of geometry from this file within each cell of the
+            CBA. If an attribute is specified, it must be categorical. In this
+            case, it will produce as much columns (binary) in the CBA matrix
+            than listed values. See parameter
+            <subset_target_attribute_values>. Note: if a vector file contains
+            several target attributes, it is necessary to run as many times the
+            <add_layer> function than target attributes of this file.
+        subset_target_attribute_values: List of values of interest for the
+            target column, in the case where a specific categorical target
+            has been specified. It allows filtering a subset of relevant
+            values.
+        target_name: Name of the produced boolean column when using all
+            geometries to dummify the data.
+
+    Returns: triplet
+        -> dummification: Boolean indicating the dummification (processing
+            of categorical attribute)
+        -> target: Targeted variable, can be artificial when only geometries
+            are considered.
+        -> identified_values: Subset of values of interest, possibly artificial
+            when only geometries are considered.
+    """
+
+    dummification = False
+    if column == "":
+        # Case where we use geometries (e.g. a point file for mineralization)
+        # we add a column filled with 1 (presence).
+        column = target_name
+        subset_target_attribute_values = [1]
+        geodata[column] = int(1)
+    else:
+        # Case of a symbolic attribute with identified subset of values of
+        # interest.
+        dummification = True
+        if subset_target_attribute_values == []:
+            subset_target_attribute_values = geodata[column].unique()
+    assert column in geodata.columns
+    identified_values = pd.Series(subset_target_attribute_values)
+    assert identified_values.isin(geodata[column]).all()
+    # FORBIDDEN AT THIS TIME as not managed !!!
+    # assert target == "" or len(attribut_values) != 0
+    assert len(subset_target_attribute_values) != 0
```

Comment:
So far we have not used asserts in any other place expect tests. These could be refactored using if-statements and then raising a custom exception as in our other functions.